Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc

```diff
@@ -213,6 +214,45 @@ export function runQTest(args: QTestArguments): Result {
         )
     );
     const qTestLogsDir: StaticDirectory = result.getOutputDirectory(logDir);
+
+    // If code coverage is enabled, start a pip that will perform coverage file upload.
+    if (qCodeCoverageEnumType === "DynamicCodeCov"){
+        let coverageLogDir = args.coverageLogsDir || Context.getNewOutputDirectory("coverageLogs");
+        let coverageConsolePath = p`${coverageLogDir}/coverageUpload.stdout`;
+        let coverageSandboxDir = Context.getNewOutputDirectory("coverageSandbox");
+
+        Debug.writeLine("Obtained coverageDirectory as " + qTestLogsDir);
+        Debug.writeLine("Obtained logs dirs as " + coverageLogDir);
+
+        let commandLineArgsForUploadPip: Argument[] = [
+            Cmd.option("--qTestLogsDir ", Artifact.output(coverageLogDir)),
+            Cmd.option("--qTestContextInfo ", Artifact.input(qTestContextInfoPath)),
+            Cmd.option("--coverageDirectory ", Artifact.input(qTestLogsDir)),
+            Cmd.option("--sandbox ", Artifact.none(coverageSandboxDir)),
+            Cmd.option("--qTestBuildType ", args.qTestBuildType),
+            Cmd.option("--qtestPlatform ", qTestPlatformToString(args.qTestPlatform))
+        ];
+
+        Transformer.execute(
+            Object.merge<Transformer.ExecuteArguments>(
+                {
+                    tool: args.qTestTool ? args.qTestTool : qTestTool,
+                    tags: tags,
+                    description: "QTest Coverage Upload",
+                    arguments: commandLineArgsForUploadPip,
+                    consoleOutput: coverageConsolePath,
+                    workingDirectory: coverageSandboxDir,
+                    tempDirectory: tempDirectory,
+                    weight: args.weight,
+                    disableCacheLookup: true,
+                    additionalTempDirectories : [coverageSandboxDir],

```

Comment:
>coverageSandboxDir [](start = 49, length = 18)

Curious why log uploader needs a sandbox? Does this imply we are creating one more copy of all the binaries?

Reply:
No. Nothing's copied into that sandbox prior to launching the upload pip. If the upload pip itself doesn't write anything in there you could remove it from the additionalTempDirectories

Reply:
Done.